src/rewrite.rs

```diff
@@ -39,4 +40,18 @@ impl<'a> RewriteContext<'a> {
             block_indent: self.block_indent + self.config.tab_spaces,
         }
     }
+
+    pub fn snippet(&self, span: Span) -> Option<String> {
+        self.codemap.span_to_snippet(span).ok()
+    }
+}
+
+impl<'v> FmtVisitor<'v> {
+    pub fn mk_context(&self) -> RewriteContext<'v> {
```

Comment:
There is already a `get_context` method on `FmtVisitor`, can we use that instead?